apps/preview/next/pages/showcases/OrderSummary/OrderSummary.tsx

```diff
@@ -23,68 +26,138 @@ export default function OrderSummary() {
 
   const totalPrice = () => itemsSubtotal() + promoCode;
 
-  const checkPromoCode = () =>
-    inputValue.toUpperCase() === 'VSF2020' ? setPromoCode(-100) : alert('Your promo code is not valid');
+  const checkPromoCode = () => {
+    if ((promoCode === -100 && inputValue.toUpperCase() === 'VSF2020') || !inputValue) return;
+    if (inputValue.toUpperCase() === 'VSF2020') {
+      setPromoCode(-100);
+      setAddedCodeAlert(true);
+      setTimeout(() => setAddedCodeAlert(false), 5000);
```

Comment:
React state setters put in timeout are not so simple. This code may lead to unexpected behaviours, because state setter will be called after 5 sec even though the component gets unmounted. You must always remember to cancel timeouts when component unmounts. Luckily, there are utils like [`useTimeoutFn`](https://github.com/streamich/react-use/blob/master/docs/useTimeoutFn.md), that would de this for us automatically

Reply:
Interesting

Reply:
https://github.com/vuestorefront/storefront-ui/pull/2627/commits/ca7074f507d51c906d126d3d40171dbaea23ce30